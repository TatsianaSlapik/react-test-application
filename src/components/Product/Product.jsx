import React from "react";
import ButtonCart from "../ButtonCart/ButtonCart";
import "./Product.scss";

const AboutTheProduct = {
  NAME: "Name",
  DESCRIPTION: "Description",
  PRICE: "Price",
};

export default function Product({ name, description, image, price }) {
  return (
    <div className="product">
      <img className="product_img" src={image} alt="photo" />
      <div>
        <h2 className="product_name">
          {AboutTheProduct.NAME} : {name}
        </h2>
        <h3 className="product_price">
          {AboutTheProduct.PRICE}:{price}
        </h3>
        <p className="product_ description">
          {AboutTheProduct.DESCRIPTION}: {description}
        </p>

        <ButtonCart />
      </div>
    </div>
  );
}
