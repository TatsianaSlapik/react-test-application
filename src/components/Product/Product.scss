.product {
  width: 270px;
  margin: 10px;
  height: 445px;
  position: relative;

  .product_img {
    width: 180px;
    height: 250px;
    margin: 0 auto;
    display: block;
  }

  .product_name {
    font-weight: 500;
  }

  .product_price {
    font-size: 30px;
  }

  .button_container {
    position: absolute;
    bottom: 0;
  }
}
.product_view {
  height: 445px;
  position: relative;
  display: flex;
  justify-content: space-evenly;

  .product_img {
    width: 180px;
    height: 250px;
  }

  .product_content {
    padding: 0 150px;
    width: 750px;
  }

  .product_name {
    font-weight: 500;
  }

  .product_price {
    font-size: 30px;
  }

  .product_description {
    margin-top: 50px;
  }

  .button_container {
    flex: none;
    bottom: 0;
    position: absolute;
  }

  .product_price_container {
    width: 215px;
    height: 300px;
    position: relative;
  }

  @media screen and (max-width: 320px) {
    display: flex;
    margin-top: 20px;
    flex-direction: column;
    height: 100%;

    .product_content {
      padding: 0 0;
      width: 100%;
    }

    .product_img {
      width: 180px;
      height: 250px;
      margin: 0 auto;
      display: block;
    }

    .product_name {
      font-weight: 500;
      padding: 0 20px;
      text-align: center;
    }

    .product_price {
      font-size: 30px;
      text-align: center;
      margin-top: 13px;
    }

    .product_price_container {
      width: 100%;
      height: 120px;
      position: relative;
    }

    .button_container {
      position: absolute;
      bottom: 0;
      width: 100%;
    }

    .product_description {
      padding: 0 20px;
      text-align: justify;
    }
  }
}
