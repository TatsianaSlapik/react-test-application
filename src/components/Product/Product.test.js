import React from "react";
import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";

import Product from "./Product.jsx";

const mockStore = configureStore([]);

describe("Product", () => {
  let store;
  let component;

  beforeEach(() => {
    store = mockStore({
      productsList: [],
      listViewState: false,
    });

    component = renderer.create(
      <Provider store={store}>
        <Product />
      </Provider>
    );
  });

  it("Product when listViewState is true should be rendered", () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
});
