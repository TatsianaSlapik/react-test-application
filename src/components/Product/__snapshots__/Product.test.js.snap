// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product Product when listViewState is true should be rendered 1`] = `
<div
  className="product"
>
  <img
    alt="photo"
    className="product_img"
  />
  <div
    className="product_content"
  >
    <h2
      className="product_name"
    />
    <h3
      className="product_price"
    >
       â‚½
    </h3>
  </div>
  <div
    className="button_container"
  >
    <button
      className="button"
      onClick={[Function]}
    >
      <img
        alt="basket"
        src={
          Object {
            "canInstrument": true,
            "createTransformer": [Function],
            "getCacheKey": [Function],
            "getCacheKeyAsync": [Function],
            "process": [Function],
            "processAsync": [Function],
          }
        }
      />
      <p>
        Add to cart
      </p>
    </button>
  </div>
</div>
`;
